src/test/objectstore/DeterministicOpSequence.cc

```diff
@@ -526,6 +561,23 @@ void DeterministicOpSequence::_do_write_and_clone_range(coll_t coll,
   m_store->apply_transaction(&m_osr, std::move(t));
 }
 
+void DeterministicOpSequence::_do_write_and_merge_delete(coll_t coll,
+                                                        hobject_t& orig_obj,
+                                                        hobject_t& new_obj,
+                                                        vector<boost::tuple<uint64_t, uint64_t, uint64_t>> move_info,
+                                                        bufferlist& bl)
+{
+  ObjectStore::Transaction t;
+
+  note_txn(&t);
+  for (unsigned i = 0; i < move_info.size(); ++i) {
+    uint64_t srcoff = move_info[i].get<0>();
+    t.write(coll, ghobject_t(orig_obj), srcoff, bl.length(), bl);
```

Comment:
You mean , write and move api happening separately some times ? 
I have followed how clone and other apis work here.
So, we can probably keep it as is for now.